bsi/src/main/java/org/roaringbitmap/bsi/longlong/Roaring64NavigableMapSliceIndex.java

```diff
@@ -0,0 +1,622 @@
+  }
+
+  /**
+   * GetValue gets the value at the column ID.  Second param will be false for non-existence values.
+   */
+  public Pair<Long, Boolean> getValue(long columnId) {
+    boolean exists = this.ebM.contains(columnId);
+    if (!exists) {
+      return Pair.newPair(0L, false);
+    }
+
+    return Pair.newPair(valueAt(columnId), true);
+  }
+
+
+  private void clear() {
+    this.maxValue = 0;
+    this.minValue = 0;
+    this.ebM = null;
+    this.bA = null;
+  }
+
+  /**
+   * Serialize this bitmapsliceindex(bsi).
+   * <p>
+   * There is no specification for now: it may change from one java version to
+   * another. This serialization format is compatible with bsi of roaring(Go)
+   * <p>
+   * Consider calling {@link #runOptimize} before serialization to improve compression.
+   * <p>
+   * The current bsi is not modified.
+   *
+   * @param output the DataOutput stream
+   * @throws IOException Signals that an I/O exception has occurred.
+   */
+  public void serialize(DataOutput output) throws IOException {
+    this.ebM.serialize(output);
+    for (Roaring64NavigableMap rb : this.bA) {
+      rb.serialize(output);
+    }
+  }
+
+  public void serialize(ByteBuffer buffer) throws IOException {
+    this.ebM.serialize(buffer);
+    for (Roaring64NavigableMap rb : this.bA) {
+      rb.serialize(buffer);
+    }
+  }
+
+  /**
+   * Deserialize (retrieve) this bitmapsliceindex(bsi).
+   * <p>
+   * There is no specification for now: it may change from one java version to
+   * another.
+   * <p>
+   * The current bsi is overwritten.
+   *
+   * @param in the DataInput stream
+   * @throws IOException Signals that an I/O exception has occurred.
+   */
+  public void deserialize(DataInput in) throws IOException {
+    this.clear();
+
+    // read ebm
+    Roaring64NavigableMap ebm = new Roaring64NavigableMap();
+    ebm.deserialize(in);
+    this.ebM = ebm;
+
+    List<Roaring64NavigableMap> baList = new ArrayList<>();
+    try( DataInputStream is = new DataInputStream((InputStream) in); ){
+      while (is.available() > 0) {
+        Roaring64NavigableMap rb = new Roaring64NavigableMap();
+        rb.deserialize(in);
+        baList.add(rb);
+      }
+    }
+    this.bA = baList.toArray(new Roaring64NavigableMap[0]);
+  }
+
+  public void deserialize(ByteBuffer buffer) throws IOException {
+    this.clear();
+    Roaring64NavigableMap ebm = new Roaring64NavigableMap();
+    ebm.deserialize(buffer);
+    this.ebM = ebm;
+    // read back
+    buffer.position(buffer.position() + (int) ebm.serializedSizeInBytes());
+    List<Roaring64NavigableMap> baList = new ArrayList<>();
+    while (buffer.hasRemaining()) {
+      Roaring64NavigableMap rb = new Roaring64NavigableMap();
+      rb.deserialize(buffer);
+      baList.add(rb);
+      buffer.position(buffer.position() + (int) rb.serializedSizeInBytes());
+    }
+    this.bA = baList.toArray(new Roaring64NavigableMap[0]);
+  }
+
+  public int serializedSizeInBytes() {
```

Comment:
Should return a long, as in `org.roaringbitmap.longlong.ImmutableLongBitmapDataProvider#serializedSizeInBytes`

Reply:
ok, should be fixed